app/models/pawn.rb

```diff
@@ -1,2 +1,43 @@
 class Pawn < Piece
+  def valid_move?(x_end, y_end)
+    diff_x = (horizontal_position - x_end).abs
+    diff_y = y_end - vertical_position if color == 'white'
+    diff_y = vertical_position - y_end if color == 'black'
+
+    return false if obstructed?(x_end, y_end) || did_not_move?(x_end, y_end)
+    return true if vertical_or_diagonal(x_end, y_end, diff_x, diff_y)
+    false
+  end
+
+  private
+
+  def vertical_or_diagonal(x_end, y_end, diff_x, diff_y)
+    vertical_move(x_end, y_end, diff_x, diff_y) || diagonal_move(x_end, y_end, diff_x, diff_y)
+  end
+
+  def vertical_move(x_end, y_end, diff_x, diff_y)
+    !piece_in_front?(x_end, y_end, diff_x, diff_y) && move_one_or_two(diff_x, diff_y)
+  end
+
+  def diagonal_move(x_end, y_end, diff_x, diff_y)
+    diff_x == 1 && diff_y == 1 && diagonal_capture?(x_end, y_end)
+  end
+
+  def diagonal_capture?(x_end, y_end)
+    return true if color == 'white' && game.pieces.where(horizontal_position: x_end, vertical_position: y_end, color: 'black').present?
+    return true if color == 'black' && game.pieces.where(horizontal_position: x_end, vertical_position: y_end, color: 'white').present?
+  end
```

Comment:
If both lines 27 and 28 are false, then this method will return nil. We need a 'false' on the last line to return false as the default case.

Or you could just have: 
`color == 'black' && game.pieces.where(horizontal_position: x_end, vertical_position: y_end, color: 'white').present?`
as the last line since if it's not true, it will evaluate to false and false will be returned.